### PR TITLE
add score bumping with query and do hashcode customized

### LINKED ISSUE #394: Score bump with query
Currently the score bumping is based on the selection of a certain result. However, I would think it would be better to only bump score when not only the result has been matched, but also the query has been match.
This will help users to separate the use of query for different high score results that both can be matched with separated query.

For example, I have a program named WeiXin: Wechat, and another one named Windows Terminal. When I want WeiXin with wx, it will be fine, but when I want Windows Terminal with acronym wt, it will sometimes move Weixin higher than Windows Terminal.

### PR DESCRIPTION
**Pull request by taooceros**
Only bump score when not only the result has been matched, but also the query has been match.
This will help users to separate the use of query for different high score results that both can be matched with separated query.

close #394

### CODE REVIEW

File: Flow.Launcher/Storage/UserSelectedRecord.cs [hunk 1]
```diff
@@ -8,16 +10,42 @@ namespace Flow.Launcher.Storage
...
+                }
+
+                foreach (var item in result.Title)
+                {
+                    hashcode = hashcode * 31 + item;
+                }
+
+                foreach (var item in result.SubTitle)
+                {
+                    hashcode = hashcode * 31 + item;
```
Comment (jjw24) [inline Flow.Launcher/Storage/UserSelectedRecord.cs:39]: 31?

Comment (taooceros) [reply Flow.Launcher/Storage/UserSelectedRecord.cs:39]: checkout https://stackoverflow.com/a/5155015
31 will have a better performance without information loss when int overflow

Comment (jjw24) [reply Flow.Launcher/Storage/UserSelectedRecord.cs:39]: Yeah, let's give it a bit of description, set it as a descriptive variable and reference that article so in the future we know where and why the logic is constructed

**Comment by jjw24** [comment]
Could you please change 31 to a variable

**Comment by taooceros** [comment]
> Could you please change 31 to a variable

Sure

**Review by jjw24 — APPROVED** [review]

**Pull request merged**